### DISCUSSION TITLE
Is it possible to tell a afterChange collection hook is triggered by save draft, publish, or unpublish in ?

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#1480] [Roadmap]
We want to call webhooks to invalidate our site pages, but found it hard to tell `save draft` apart from the other two.

**Comment by jacobsfletch** [comment]
Hey @Stupidism in your `afterChange` hook, you should be able to read the `_status` of the document to determine whether its published or draft.

**Reply by jmikrut** [reply]
One other note is that we could easily expose the `draft` argument to all collection hooks. We will mark this as `planned`.

**Reply by jmikrut** [reply]
On second thought, I believe that this is redundant. The `_status` field is always going to be `draft` if hooks are fired with `draft: true`. So I don't think there's anything to do here after all.

**Reply by adamalfredsson** [reply]
What I'm struggling with here is that it's difficult to infer from `_status` if a document was unpublished. I want to trigger a static frontend build when an editor publishes or unpublishes a document, but not when saving drafts. How would I do this?

I wouldn't mind seeing some explicit hooks, e.g. `afterPublish`, `afterUnpublish`

**Comment by jacobsfletch** [comment]
Related: https://github.com/payloadcms/payload/discussions/1996
